admin/worker/delete_unsued_assets.go

```diff
@@ -0,0 +1,80 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+const _defaultPageSize = 20
+
+func (w *Worker) deleteUnusedAssets(ctx context.Context) error {
+	// We skip unused assets created in last 15 minutes to prevent race condition
+	// where somebody just created an asset but is yet to use it
+	createdBefore := time.Now().Add(-15 * time.Minute)
+	for {
+		// 1. Fetch unused assets
+		assets, err := w.admin.DB.FindUnusedAssets(ctx, createdBefore, _defaultPageSize)
+		if err != nil {
+			return err
+		}
+		if len(assets) == 0 {
+			return nil
+		}
+		createdBefore = assets[len(assets)-1].CreatedOn
+		ids := make([]string, len(assets))
```

Comment:
I would suggest simplifying the logic to always delete the full page fetched. And if any file delete fails with a different error than "not exists", can just return an error from the job (we will be alerted on it, and if the error was ephemeral, it will just be retried on next job invocation).

That avoids pagination by created on and will avoid values sticking around in the DB for too long.

Reply:
Sure. My thinking was that a few continuous failures can end up building lot of unused assets in the table and it will be good to keep deleting at least the successful deletes. 
When issuing 100 network calls it feels somewhat weird to only take further action when all 100 pass.

> will avoid values sticking around in the DB for too long.

No right if we return without deleting anything than all values stick in the DB till the next successful run.

Reply:
> When issuing 100 network calls it feels somewhat weird to only take further action when all 100 pass.

Normally I would agree, but in this case all the calls have the same type and go to the same service, so any other error than "not exists" should be a service-wide error, not a resource-specific error. So it is okay to keep it simple and just retry the whole batch at a later point.